sstmac/common/event_scheduler.h

```diff
@@ -513,11 +512,10 @@ using SubComponentParent = MacroBaseComponent;
 class Component : public ComponentParent
 {
  public:
-  virtual ~Component() {}
-
-  virtual void setup(); //needed for SST core compatibility
+  ~Component() {}
```

Comment:
Doesn't this still need to be virtual? I can always delete a type of `Component*` and then this wouldn't be a virtual destructor anymore.

Reply:
No because MacroBaseComponent has a virtual destructor. unless I'm missing something.